### PR TITLE
Cleanup misc System.Reflection tests

### PR DESCRIPTION
**Pull request by hughbe**
Small stuff but another in my System.Reflection backlog
/cc @stephentoub

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test this please (PR submitted when there was a build break)

File: src/System.Reflection/tests/ModuleTests.cs [hunk 1]
```diff
@@ -41,24 +42,20 @@ public static void FullyQualifiedNameTest(TypeInfo typeInfo)
...
         [InlineData(typeof(EnumAttr), PublicEnum.Case1, "EnumAttrSimple")]        
-        public static void CustomAttributesTest<CtorArg, NamedArg>(Type attrType, CtorArg expectedCtorValue, NamedArg expectedNamedValue)
+        public void CustomAttributes<CtorArg, NamedArg>(Type attrType, CtorArg expectedCtorValue, NamedArg expectedNamedValue)
         {
             Module module = typeof(ModuleTest).GetTypeInfo().Module;
-            var customAttrs = module.CustomAttributes.GetEnumerator();
 
             CustomAttributeData attribute = module.CustomAttributes.SingleOrDefault(a => a.AttributeType.Equals(attrType));
-
-            Assert.NotNull(attribute);
```
Comment (stephentoub) [inline src/System.Reflection/tests/ModuleTests.cs]: Why was this removed?

File: src/System.Reflection/tests/CoreCLR/Assembly/AssemblyTests.CoreCLR.cs [hunk 2]
```diff
@@ -12,21 +12,18 @@ namespace System.Reflection.Tests
...
-        public void CurrentAssemblyHasLocation()
+        public void CurrentLocation_HasLocaton()
         {
-            string location = ThisAssembly().Location;
-
-            Assert.NotNull(location);
-            Assert.NotEmpty(location);
-
+            string location = GetExecutingAssembly().Location;
+            Assert.False(string.IsNullOrEmpty(location));
```
Comment (hughbe) [inline src/System.Reflection/tests/CoreCLR/Assembly/AssemblyTests.CoreCLR.cs]: I've replaced this with `Assert.NotEmpty`. This checks for null explicitly as well as the empty case, differentiating between the two. Same goes for `Assert.Empty`.

See the code [here](https://github.com/xunit/assert.xunit/blob/bbece92b18661cc18ede9d72b28648466f9a1709/CollectionAsserts.cs#L182)

File: src/System.Reflection/tests/CoreCLR/Assembly/AssemblyTests.CoreCLR.cs [hunk 3]
```diff
@@ -35,40 +32,34 @@ public void CurrentAssemblyHasLocation()
...
         [Fact]
-        public void LoadFromStreamHasEmptyLocation()
+        public void LoadFromStream_Location_IsEmpty()
         {
             Assembly assembly = new TestStreamLoadContext().LoadFromAssemblyName(new AssemblyName("TinyAssembly"));
-            string location = assembly.Location;
-
-            Assert.NotNull(location);
-            Assert.Empty(location);
+            Assert.Empty(assembly.Location);
```
Comment (hughbe) [inline src/System.Reflection/tests/CoreCLR/Assembly/AssemblyTests.CoreCLR.cs]: See previous comment :)

File: src/System.Reflection/tests/CoreCLR/Assembly/AssemblyTests.CoreCLR.cs [hunk 4]
```diff
@@ -35,40 +32,34 @@ public void CurrentAssemblyHasLocation()
...
             MethodInfo entryPoint = typeof(TestExe).GetTypeInfo().Assembly.EntryPoint;
             Assert.NotNull(entryPoint);
             Assert.Equal(42, entryPoint.Invoke(null, null));
         }
 
-        private static Assembly ThisAssembly()
-        {
-            return typeof(AssemblyTests).GetTypeInfo().Assembly;
-        }
+        private static Assembly GetExecutingAssembly() => typeof(AssemblyTests).GetTypeInfo().Assembly;
```
Comment (hughbe) [inline src/System.Reflection/tests/CoreCLR/Assembly/AssemblyTests.CoreCLR.cs]: All the other files in the main test project which have this method are called GetExecutingAssembly

File: src/System.Reflection/tests/ModuleTests.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Reflection/tests/ModuleTests.cs]: Not sure - I assumed an NRE would be enough if this failed.

I addressed your comment by replacing `SingleOrDefault` with `Single`, so that we get an explicit error message detailing that we couldn't find the attribute mentioned

**Comment by hughbe** [comment]
Test Innerloop Windows_NT Release Build and Test
